core/src/main/java/org/fao/geonet/kernel/security/SecurityProviderConfiguration.java

```diff
@@ -0,0 +1,99 @@
+/*
+ *  Copyright (C) 2014 GeoSolutions S.A.S.
+ *  http://www.geo-solutions.it
+ *
+ *  GPLv3 + Classpath exception
+ *
+ *  This program is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package org.fao.geonet.kernel.security;
+
+import org.fao.geonet.ApplicationContextHolder;
+import org.fao.geonet.exceptions.BadInputEx;
+import org.fao.geonet.exceptions.BadParameterEx;
+
+import java.util.Map;
+
+/**
+ * Some basic configuration info to be used by all security
+ *
+ */
+public interface SecurityProviderConfiguration {
+
+	public enum LoginType {
```

Comment:
Does this configuration affects `LDAP` or `CAS` providers?

Reply:
It was mostly used for Shibboleth for the SSO parameters.  For all cases it would check for the existence of the bean and if the bean was not found then then it continued processing as it did before.  So unless LDAP or CAS was using Shibboleth configuration options, they should not be affected by these changes.

Once this PR is approved then it would probably be a good idea if LDAP or CAS also used this class if there is a need to support common configurations for SSO.